### ISSUE TITLE
SignalXY: Too many markers drawn in step display mode

### ISSUE THREAD

**Issue by grabul** [ScottPlot/ScottPlot#1896]
After version > 4.1.33 if StepDisplay = true every point to draw a line to is now marked with a marker (if marker size > 0).

![image](https://user-images.githubusercontent.com/80955161/174806763-057586be-4532-4e59-bee5-288af085bf37.png)

Possible fix in SignalPlotXYGeneric.cs:

add line

```     
        PointF[] PointsToMark = (PointF[])PointsToDraw.Clone();
```

before 

```
        if (StepDisplay)
            PointsToDraw = GetStepPoints(PointsToDraw);
```

and replace variable PointsToDraw with PointsToMark in the section below where the drawing of the markers is done

```
      if (PointsToMark.Length > 1)
      {
          float dataSpanXPx = PointsToMark[PointsToMark.Length - 1].X - PointsToMark[0].X;
          float markerPxRadius = .3f * dataSpanXPx / PointsToMark.Length;
          markerPxRadius = Math.Min(markerPxRadius, MarkerSize / 2);
          float scaledMarkerSize = markerPxRadius * 2;

          if (markerPxRadius > .3)
          {
              // skip not visible before and after points
              var PointsWithMarkers = PointsToMark
                                      .Skip(PointBefore.Length)
                                      .Take(PointsToDraw.Length - PointBefore.Length - PointAfter.Length)
                                      .ToArray();

              MarkerTools.DrawMarkers(gfx, PointsWithMarkers, MarkerShape, scaledMarkerSize, MarkerColor, MarkerLineWidth);
          }
      }
```

**Comment by swharden** [comment]
Thanks for pointing this out @grabul!

Note to self, relevant code is here:

https://github.com/ScottPlot/ScottPlot/blob/084f998e236e10421cf45224cdb090de5138bfcc/src/ScottPlot4/ScottPlot/Plottable/SignalPlotXYGeneric.cs#L219-L221

https://github.com/ScottPlot/ScottPlot/blob/084f998e236e10421cf45224cdb090de5138bfcc/src/ScottPlot4/ScottPlot/Plottable/SignalPlotXYGeneric.cs#L248-L266

**Comment by swharden** [comment]
Thanks for opening this issue grabul! This should be solved in #1896 👍 

before | after
---|---
![image](https://user-images.githubusercontent.com/4165489/174910470-43a7e3f1-04e5-40d1-b683-bad22685749d.png)|![image](https://user-images.githubusercontent.com/4165489/174910455-06310e50-0fa4-4396-bf86-95b3d42d03ba.png)
